### PR TITLE
added option for mounting unlocked fs for non root users and support for cryptdevice kernel parameter

### PR DESCRIPTION
**Pull request by stuffo**
also disabled verbose expect output

### CODE REVIEW

File: src/hooks/ykfde [hunk 1]
```diff
@@ -120,13 +120,36 @@ ykfde_do_it() {
...
+    local cryptargs cryptopt cryptoptions
 
     [ "$DBG" ] && echo "$0:"
 
     [ "$DBG" ] && echo " > Reading YKFDE configuration file."
     . "$YKFDE_CONFIG_FILE" || { ykfde_err 001 "Failed to read the YKFDE configuration file '$YKFDE_CONFIG_FILE'"; return 1; }
 
-    [ -z "$YKFDE_DISK_UUID" ] || [ -z "$YKFDE_LUKS_NAME" ] && { ykfde_err 002 "'$YKFDE_CONFIG_FILE' Please provide YKFDE_DISK_UUID /and/ YKFDE_LUKS_NAME."; return 1; }
+    # no settings in config, try to pull it from kernel cmdline (analog to encrypt hook)
+    if [ -z "$YKFDE_DISK_UUID" -o -z "$YKFDE_LUKS_NAME" ] ; then
```
Comment (Vincent43) [inline src/hooks/ykfde]:
```
In ykfde line 131:
    if [ -z "$YKFDE_DISK_UUID" -o -z "$YKFDE_LUKS_NAME" ] ; then
                               ^-- SC2166: Prefer [ p ] || [ q ] as [ p -o q ] is not well defined.

```

File: src/hooks/ykfde [hunk 2]
```diff
@@ -120,13 +120,36 @@ ykfde_do_it() {
...
+        if [ -n "${cryptdevice}" ] ; then
+            IFS=: read YKFDE_LUKS_DEV YKFDE_LUKS_NAME cryptoptions <<EOF
```
Comment (Vincent43) [inline src/hooks/ykfde]:
```
In ykfde line 133:
            IFS=: read YKFDE_LUKS_DEV YKFDE_LUKS_NAME cryptoptions <<EOF
            ^-- SC2162: read without -r will mangle backslashes.
```

File: src/hooks/ykfde [hunk 3]
```diff
@@ -120,13 +120,36 @@ ykfde_do_it() {
...
+$cryptdevice
+EOF
+            for cryptopt in ${cryptoptions//,/ }; do
```
Comment (Vincent43) [inline src/hooks/ykfde]:
```
In ykfde line 136:
            for cryptopt in ${cryptoptions//,/ }; do
                            ^-- SC2169: In dash, string replacement is not supported.
```

File: src/ykfde-open [hunk 4]
```diff
@@ -86,12 +91,19 @@ if [ "$(id -u)" -eq 0 ]; then
...
-    expect -exact "Passphrase: "
-    send -- "${_passphrase}\\r"
-    expect eof
+	set timeout -1
+	log_user 0
+	spawn udisksctl unlock -b "$YKFDE_LUKS_DEV"
+	match_max 100000
+	expect -exact "Passphrase: "
+	send -- "${_passphrase}\\r"
+	expect -re "Unlocked .* as (.*).\r\n" {
```
Comment (Vincent43) [inline src/ykfde-open]:
```
In ykfde-open line 100:
        expect -re "Unlocked .* as (.*).\r\n" {
                                        ^-- SC1117: Backslash is literal in "\r". Prefer explicit escaping: "\\r".
                                          ^-- SC1117: Backslash is literal in "\n". Prefer explicit escaping: "\\n".
```

**Comment by stuffo** [comment]
done :wink:

**Comment by Vincent43** [comment]
Thanks. Very nice additions.

**Comment by agherzan** [comment]
Good job!

**Pull request merged**